core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraXCameraUseCase.kt

```diff
@@ -654,6 +660,51 @@ constructor(
         } ?: this
     }
 
+    private fun CameraAppSettings.tryApplyFrameRateConstraints(): CameraAppSettings {
+        return systemConstraints.perLensConstraints[cameraLensFacing]?.let { constraints ->
+            with(constraints.supportedFixedFrameRates) {
+                val newTargetFrameRate = if (contains(targetFrameRate)) {
+                    targetFrameRate
+                } else {
+                    TARGET_FPS_AUTO
+                }
+
+                this@tryApplyFrameRateConstraints.copy(
+                    targetFrameRate = newTargetFrameRate
+                )
+            }
+        } ?: this
+    }
+
+    private fun CameraAppSettings.tryApplyStabilizationConstraints(): CameraAppSettings {
+        val currentFrameRate = currentSettings.value?.targetFrameRate ?: return this
+
+        return systemConstraints.perLensConstraints[cameraLensFacing]?.let { constraints ->
+            with(constraints.supportedStabilizationModes) {
+                val newVideoStabilization = if (contains(SupportedStabilizationMode.HIGH_QUALITY) &&
+                    (currentFrameRate == TARGET_FPS_AUTO || currentFrameRate == TARGET_FPS_30)
```

Comment:
This should still be currentFrameRate <= TARGET_FPS_60 if we want to match `shouldVideoBeStabilized`.

Reply:
oh whoops, i think i got a  bit mixed up there, thanks for pointing out :-)